numpy/lib/function_base.py

```diff
@@ -3541,7 +3541,7 @@ def _percentile(a, q, axis=None, out=None,
     elif interpolation == 'higher':
         indices = ceil(indices).astype(intp)
     elif interpolation == 'midpoint':
-        indices = floor(indices) + 0.5
+        indices = 0.5 * (floor(indices) + ceil(indices))
```

Comment:
Frankly, I am not sure :(, it might be my error for thinking about it back the. IIRC, it might be there because it is what matlab uses when it draws boxplots (this statement may be completly wrong).
I am happy about corrections, even deprecations if it is found too weird...